### ISSUE TITLE
Keep program running while server_http instance is listening?

### ISSUE THREAD

**Issue by daniel-at-yoke** [davisking/dlib#725]
Maybe not an issue, as much as a question, but what's a good way to keep a program running as long as a `server_http` instance is listening? The server example uses a infinite loop to listen for (and discard) data on stdin, which just makes the example "idle" with high CPU usage.

Could put a `sleep` in there, but I imagine it'd be better to somehow just `wait`, if the program is intended to run until explicitly killed (as is the goal in my case). Right now I'm just creating a mutex and signaler in `main`, and `wait`ing indefinitely that way. It works, but feels hacky, since it's got nothing to do with the server running, really.

As far as I can tell there's no way to "directly" wait for the server, or am I mistaken? (Should add: I'm still a C++ novice)

**Comment by davisking** [comment]
Which example shows an infinite loop?

**Comment by daniel-at-yoke** [comment]
Hmm, none it seems. Must have been mistaken. I was thinking of [this example](https://github.com/davisking/dlib/blob/master/examples/server_http_ex.cpp), which just looks at `cin.get()` and which I adapted. I guess I wrapped that `get` in a `while(true)` to have it _not_ stop on stdin input but just keep waiting. However, even adding that to the plain example, I'm not getting the high CPU usage - which is what you'd expect. However... when I attach Xcode's debugger, the program starts idling at 100%... thanks, Xcode, for sending me on a while goose chase.

So apologies; I was digging in the wrong place.

Still, as a matter of curiosity, is there a better way to wait for the server, instead of waiting for something unrelated like stdin? Functionally, `main` can wait forever on anything, but semantically I feel like it'd make more sense to wait for the server instance. I suppose I can add something in a derived class, of course, just wondering if I missed anything in base implementation.
